net/url_request/url_fetcher_response_writer.h

```diff
@@ -92,6 +101,35 @@ class NET_EXPORT URLFetcherStringWriter : public URLFetcherResponseWriter {
   DISALLOW_COPY_AND_ASSIGN(URLFetcherStringWriter);
 };
 
+#if defined(IN_MEMORY_UPDATES)
+// Memory-conscious URLFetcherResponseWriter implementation for a "large"
+// std::string. The string's capacity is preallocated to the size of the content
+// and the data can be "moved" out.
+// Similar to cobalt::loader::URLFetcherStringWriter but with a different
+// preallocation strategy.
+class NET_EXPORT URLFetcherLargeStringWriter : public URLFetcherResponseWriter {
```

Comment:
Couldn't this live somewhere in cobalt/updater rather than in upstream code ?

Reply:
It could, but I think there's a benefit to having it near `URLFetcherStringWriter`. They are kind of sibling classes and their function implementations and unit tests look similar, so from a readability perspective I think it helps to have them near one another. But lmk if you disagree or think the cost of the preprocessor guards outweighs that.

Reply:
I think i disagree on this one. It's a Cobalt developed piece of code, it should live in /cobalt, not in upstream.

Reply:
As we agreed to leave the implementation as is, leave this here as well.